### PR TITLE
feat: remove implementation, restructure and add augmentation and consent

### PR DESCRIPTION
**Pull request by sethyates**
## Description of this change

Remove implementation, restructure, add augmentation and add consent

## Why is this change being made?
- [ ] Chore (non-functional changes)
- [ ] Bug fix (non-breaking change that fixes an issue)
- [X] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## How was this tested? How can the reviewer verify your testing?

N/A

## Checklist

- [ ] I have added tests to cover my changes.
- [X] All new and existing tests passed.
- [X] I have evaluated the security impact of this change, and [OWASP Secure Coding Practices](https://owasp.org/www-project-secure-coding-practices-quick-reference-guide/migrated_content#) have been observed.
- [X] I have informed stakeholders of my changes.

### CODE REVIEW

File: api/dsr/v1/Augment.md [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+
+{
+  "apiVersion": "dsr/v1",
+  "kind": "AugmentRequest",
+  "metadata": {
+    "uid": "22880925-aac5-42f9-a653-cb6921d361ff",
+    "tenant": "axonic"
+  },
+  "request": {
+    "kind": "DeleteRequest",
```
Comment (neeth04) [inline api/dsr/v1/Augment.md:23]: This will be "AugmentRequest" kind?

File: api/dsr/v1/Correction.md [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+| *event.expectedCompletionTimestamp* | no        | The UNIX time stamp at which the Data Subject Request is expected to be completed                                                                                                           |
+| *event.results*                     | no        | Array of [Callbacks](README.md#Callback) that can be used to download the contents requested                                                                                                |
+| *event.requestID*                   | no        | The request ID known to the destination system                                                                                                                                              |
+| *event.results*                     | no        | Array of [Callbacks](README.md#Callback) that can be used to download the contents requested                                                                                                |
```
Comment (neeth04) [inline api/dsr/v1/Correction.md:163]: redundant

File: api/dsr/v1/Delete.md [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+| *event.expectedCompletionTimestamp* | no        | The UNIX time stamp at which the Data Subject Request is expected to be completed                                                                                                           |
+| *event.results*                     | no        | Array of [Callbacks](README.md#Callback) that can be used to download the contents requested                                                                                                |
+| *event.requestID*                   | no        | The request ID known to the destination system                                                                                                                                              |
+| *event.results*                     | no        | Array of [Callbacks](README.md#Callback) that can be used to download the contents requested                                                                                                |
```
Comment (neeth04) [inline api/dsr/v1/Delete.md:169]: redundant

File: api/dsr/v1/RestrictProcessing.md [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+| *event.expectedCompletionTimestamp* | no        | The UNIX time stamp at which the Data Subject Request is expected to be completed                                                                                                           |
+| *event.results*                     | no        | Array of [Callbacks](README.md#Callback) that can be used to download the contents requested                                                                                                |
+| *event.requestID*                   | no        | The request ID known to the destination system                                                                                                                                              |
+| *event.results*                     | no        | Array of [Callbacks](README.md#Callback) that can be used to download the contents requested                                                                                                |
```
Comment (neeth04) [inline api/dsr/v1/RestrictProcessing.md:170]: remove redundant entry

File: api/dsr/v1/README.md [hunk 5]
```diff
@@ -0,0 +1,104 @@
+# Data Subject Requests
+
+## Request Types
+
+* [Access](Access.md)
+* [Correction](Correction.md)
+* [Delete](Delete.md)
+* [Restrict Processing](RestrictProcessing.md)
```
Comment (neeth04) [inline api/dsr/v1/README.md:8]: should we also add "AugmentRequest" as a type?

File: api/dsr/v1/Access.md [hunk 6]
```diff
@@ -0,0 +1,180 @@
...
+Authorization: Bearer $auth
+
+{
+  "apiVersion": "dsr/v1",
+  "kind": "AccessRequest",
+  "metadata": {
+    "uid": "22880925-aac5-42f9-a653-cb6921d361ff",
+    "tenant": "axonic"
+  },
+  "request": {
```
Comment (neeth04) [inline api/dsr/v1/Access.md:23]: Add "kind" field as it is required?

Comment (sethyates) [reply api/dsr/v1/Access.md:23]: huh?

File: api/dsr/v1/Access.md [hunk 7]
```diff
@@ -0,0 +1,170 @@
...
+
+### Mutate identities
+
+* To add identities, return additional [Identities](README.md#Identity) in the `identities` property.
+* To replace an identity, return an [Identity](README.md#Identity) with the same `identitySpace` in the `identities` property.
+* To remove an identity, return an [Identity](README.md#Identity) with the same `identitySpace` in the `identities` property but without any `identityValue` property.
+
+### Add results
+
+To add results that should be made available to the Data Subject, return result documents in the `results` property.
```
Comment (smalexan) [inline api/dsr/v1/Access.md:142]: This is a json object now right? Not callback like example? https://github.com/ketch-com/ketch-forwarder/pull/11/files#diff-72ba22185acb3032dda0910db9ad336580b5c954ea6376c0480084b65fc74ab3R112-R119

**Review by smalexan — APPROVED** [review]

File: api/dsr/v1/Access.md [hunk 8]
```diff
@@ -127,50 +129,56 @@ Content-Type: application/json
...
+| *expectedCompletionTimestamp* | no        | The UNIX timestamp at which the Data Subject Request is expected to be completed                                                                                                            |
+| *requestID*                   | no        | The request ID known to the destination system                                                                                                                                              |
+| *results*                     | no        | Array of [Documents](README.md#Document) that can be used to download the contents requested                                                                                                |
+| *documents*                   | no        | Array of [Documents](README.md#Document) that can be used to download the contents requested                                                                                                |
+| *claims*                      | no        | Map containing additional claims that are to be added                                                                                                                                       |
```
Comment (smalexan) [inline api/dsr/v1/Access.md:179]: Need to support updating / removing claims?

File: api/dsr/v1/Access.md [hunk 9]
```diff
@@ -127,50 +129,56 @@ Content-Type: application/json
...
+| *redirectUrl*                 | no        | if the [Data Subject](README.md#Subject) should be redirected to a URL (perhaps for confirmation)                                                                                           |
+| *subject*                     | no        | The complete [Data Subject](README.md#Subject) to replace the Data Subject details or an array of [JSON Patch](https://jsonpatch.com/) specification of the Data Subject details to mutate. |
+| *identities*                  | no        | Array of [Identities](README.md#Identity) to add to the request                                                                                                                             |
```
Comment (smalexan) [inline api/dsr/v1/Access.md:182]: Need to support updating / removing identities?

File: api/dsr/v1/Access.md (hunk 8 above)
Comment (sethyates) [reply api/dsr/v1/Access.md:179]: Update by returning same key. Remove by using `null` as value

Comment (sethyates) [reply api/dsr/v1/Access.md:179]: Or you mean the text?

Comment (smalexan) [reply api/dsr/v1/Access.md:179]: That makes sense. Is there a reason why data subject should behave differently w json patch?

```| *subject* | no | The complete [Data Subject](README.md#Subject) to replace the Data Subject details or an array of [JSON Patch](https://jsonpatch.com/) specification of the Data Subject details to mutate. |```

File: api/dsr/v1/README.md [hunk 10]
```diff
@@ -62,14 +62,14 @@ The Document object can provide the data directly in the value.
...
-    "Content-Type": "application/octet-stream"
+    "Content-Type": "application/json"
   }
 }
 ```
 
 Here, the document is returned directly in the response/event using base64-encoded data. The
 `Content-Type` header is required so Ketch knows the [mime type](https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/MIME_types/Common_types)
-of data received.
+of data received. Supported content types include: `application/json`. Response data is limited to 1MB.
```
Comment (smalexan) [inline api/dsr/v1/README.md]: only

**Review by suejungshin — APPROVED** [review]

**Pull request merged**